### ISSUE TITLE
NVDA does not read the symbols inserted by the word 2013 equation editor

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4631]
**Reported by pranavlal on 2014-11-18 14:07**
NVDA does not read the symbols inserted by the word 2013 equation editor. The editor uses a font calibry Math as far as I can tell. Moreover, it does not report if alphabits in that equation are sub script or super script. To duplicate the problem, do the following.
1. launch word 2013.
2. Navigate to the insert tab in the ribbin and then to equation.
3. Insert an equation from the grid.

Try reading the equation such as that for the area of a circle. No go, the special symbols are not read.

Pranav

**Comment by bhavyashah** [comment]
@michaelDCurran @jcsteh @Qchristensen What priority do you think should be assigned to this ticket? Since the implementation cost seems to be non-trivial, I am leaning towards P3 or P4, although I have little knowledge about the feature's value and impact.

**Comment by jcsteh** [comment]
It'd be good to be able to read native Microsoft Word math using NVDA's math support. (We do support MathType, but that requires a separate product.) However, there are a few challenges/considerations here:

1. Office doesn't use MathML internally; it uses OMML. It includes an XSLT style sheet to convert between the two. We'd need to bundle an XSLT transformer to make use of this, perhaps lxml.
2. Math occupies more than one character, whereas we really want a single character stop and then the user uses NVDA/MathPlayer commands to interact with the math in a more screen reader friendly way. I'm not sure how to get around this. Perhaps all we can do is let browse mode fake the math as a single character stop. I'd be very reluctant to override the character movement commands in focus mode.
3. Microsoft have done some work to expose math speech via UIA. This doesn't expose the markup, though, which means it's completely different to NVDA's own math support and cannot support braille. Still, it's possible Microsoft will do more on this, so we should make sure we're fully aware of what's happening there before considering any implementation.

P3 for now, though there will probably be a high implementation cost, so this may get dropped to p4.

**Comment by Adriani90** [comment]
This is now fixed in #16521. @LeonarddeR it would be great if you enable that by default and make it a checkbox.

**Comment by LeonarddeR** [comment]
As long as this is not the default, I wouldn't consider this fixed. Also note that this is not listed in the change log nor in the documentation.

**Comment by CyrilleB79** [comment]
If enabling the option improves things for Word equation reading, it may be worth mentioning it in the change log as well as in the User Guide.
Re the User Guide, information should be added where the new normalization options are described, but also maybe in paragraph 7 (reading math content).

@LeonarddeR or @Adriani90, would you mind add these changes?

**Comment by LeonarddeR** [comment]
I'm happy to do so. If there's a wish to enable this by default as well as change it from a combobox into a check box, that's fine with me as well.
@seanbudd Could you chime in here?

**Comment by LeonarddeR** [comment]
I'm not sure whether everything regarding the Word equation editor is fixed, especially the superscript and subscript part. Could anyone confirm that?

**Comment by Adriani90** [comment]
Yes it reads out to me properly. Did you enable the coresponding checkboxes il document formatting settings?Von meinem iPhone gesendetAm 22.05.2024 um 17:02 schrieb Leonard de Ruijter ***@***.***>:﻿
I'm not sure whether everything regarding the Word equation editor is fixed, especially the superscript and subscript part. Could anyone confirm that?

**Comment by seanbudd** [comment]
@LeonarddeR 

>I'm happy to do so. If there's a wish to enable this by default as well as change it from a combobox into a check box, that's fine with me as well.
@seanbudd Could you chime in here?

I think this option is best as a feature flag combo-box. That way we can enable this by default in future if we wish.
I'd encourage moving this discussion elsewhere as it's offtopic

**Comment by seanbudd** [comment]
Is it possible that unicode normalization would automatically apply for equation editing? Perhaps we need a "smart" mode for normalization given the certain cases we think it should apply, like UIA.

**Comment by Qchristensen** [comment]
Just testing now in Office 365 (64-bit) Version: 16.0.18025.20160 
with NVDA 2024.4 RC1.

This seems to work for me.  Testing the suggested area of a circle and NVDA reads "A=πr^2" (Cap A = Pi R Squared)
Is anyone still having issues with this?

**Comment by Adriani90** [comment]
This works as expected when UIA for MS Word is enabled, but not when using objectModel.

**Comment by LeonarddeR** [comment]
Since UIA for Word is the default for Modern Word, I'm going to close this. Supporting this with the object model is probably not very trivial anyway.

**Comment by CyrilleB79** [comment]
Sorry @LeonarddeR, have you seen #18401?

Until a clear policy for legacy Word object model bug reports is established, I'd recommend holding off on closing such issues.
More specifically, if the third options (Close, but track via a label and meta-issue / discussion) is chosen / confirmed we shouldn't close the issues until such a label and meta-issue / discussion is created and the current issue referenced in it.
Thanks.

I'll reopen this one, at least temporarily.

**Comment by LeonarddeR** [comment]
@CyrilleB79 I have missed that. Yet I'm a bit uncertain about this particular issue, since it clearly mentions Word 2013, which is pretty out of date now. The issue is also very old itself, doesn't follow our current issue standards (i.e. it doesn't have the bug reporting template in it). Also please keep note of the fact that the issue was linked in the changelof for 2024.3.

Of course I do not want to redo the discussion in #18401, but I would rather see for this one that a new issue will be opened if there is actually a need for a fix for the object model.

**Comment by CyrilleB79** [comment]
I have tested with normalization enabled (default) in Word 2021 LTSC. The equation's symbols are read correctly: "A equals pi R 2". The structure of the equation, such as indicating that "2" is superscript, is not read as is. But I do not think it was the object of this issue.

In https://github.com/nvaccess/nvda/issues/4631#issuecomment-2993236237, @Adriani90  wrote:
> This works as expected when UIA for MS Word is enabled, but not when using objectModel.

If @Adriani90, @pranavlal or anybody else can reproduce this issue, could you please give the exact steps, options used (e.g. normalization, UIA vs legacy, etc.) as well as versions of Office, Windows and NVDA? Thanks.

**Comment by seanbudd** [comment]
did the integration of mathcat improve this at all?
